core/gallery/service.go

```diff
@@ -0,0 +1,494 @@
+			Type:                  model.Import_Pb,
+			Mode:                  pb.RpcObjectImportRequest_ALL_OR_NOTHING,
+			NoProgress:            progress == nil,
+			IsMigration:           false,
+			Params: &pb.RpcObjectImportRequestParamsOfPbParams{
+				PbParams: &pb.RpcObjectImportRequestPbParams{
+					Path:            []string{path},
+					NoCollection:    true,
+					CollectionTitle: title,
+					ImportType:      pb.RpcObjectImportRequestPbParams_EXPERIENCE,
+				}},
+			IsNewSpace: isNewSpace,
+		},
+		Origin:   origin,
+		Progress: progress,
+		IsSync:   true,
+	}
+	res := s.importer.Import(ctx, importRequest)
+
+	return res.Err
+}
+
+func (s *service) downloadZipToFile(url string, progress process.Progress) (path string, err error) {
+	if s.withUrlValidation {
+		if err = uri.ValidateURI(url); err != nil {
+			return "", fmt.Errorf("provided URL is not valid: %w. Or invalid path to file", err)
+		}
+		if !isInWhitelist(url) {
+			return "", fmt.Errorf("provided URL is not in whitelist")
+		}
+	}
+
+	var (
+		countReader *datacounter.ReaderCounter
+		size        int64
+	)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	readerMutex := sync.Mutex{}
+	defer cancel()
+	go func() {
+		counter := int64(0)
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-progress.Canceled():
+				cancel()
+			case <-time.After(time.Second):
+				readerMutex.Lock()
+				if countReader != nil && size != 0 {
+					count := countReader.Count()
+					if count > uint64(^int64(0)) {
+						count = uint64(^int64(0))
+					}
+					progress.SetDone(archiveDownloadingPercents + archiveCopyingPercents*int64(count)/size)
+				} else if counter < archiveDownloadingPercents {
+					counter++
+					progress.SetDone(counter)
+				}
+				readerMutex.Unlock()
+			}
+		}
+	}()
+
+	var reader io.ReadCloser
+	reader, size, err = getArchiveReaderAndSize(url)
+	if err != nil {
+		return "", err
+	}
+	defer reader.Close()
+	readerMutex.Lock()
+	countReader = datacounter.NewReaderCounter(reader)
+	readerMutex.Unlock()
+
+	path = filepath.Join(s.tempDirService.TempDir(), time.Now().Format("tmp.20060102.150405.99")+".zip")
+	var out *os.File
+	out, err = os.Create(path)
+	if err != nil {
+		return "", anyerror.CleanupError(err)
+	}
+	defer out.Close()
+
+	if _, err = io.Copy(out, countReader); err != nil {
+		return "", err
+	}
+
+	progress.SetDone(archiveDownloadingPercents + archiveCopyingPercents)
+	return path, nil
+}
+
+func (s *service) setupProgress() (process.Notificationable, error) {
+	progress := process.NewNotificationProcess(pb.ModelProcess_Import, s.notifications)
+	if err := s.progress.Add(progress); err != nil {
+		return nil, fmt.Errorf("failed to add progress bar: %w", err)
+	}
+	progress.SetProgressMessage("downloading archive")
+	progress.SetTotal(100)
+	return progress, nil
+}
+
+func getArchiveReaderAndSize(url string) (reader io.ReadCloser, size int64, err error) {
+	client := http.Client{Timeout: 15 * time.Second}
+	// nolint: gosec
```

Comment:
Why do we need this comment? It seems like there is no security issues? 🧐

Reply:
Actally there is. Linter worried about referencing some mal resources :see_no_evil: